node/src/accountant/scanners/mod.rs

```diff
@@ -1349,11 +1391,139 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn keeping_entries_consistent_and_aligned_is_so_important() {
+        let wallet_1 = make_wallet("abc");
+        let hash_1 = make_tx_hash(123);
+        let wallet_2 = make_wallet("def");
+        let hash_2 = make_tx_hash(345);
+        let wallet_3 = make_wallet("ghi");
+        let hash_3 = make_tx_hash(546);
+        let wallet_4 = make_wallet("jkl");
+        let hash_4 = make_tx_hash(678);
+        let pending_payables_owned = vec![
+            PendingPayable::new(wallet_1.clone(), hash_1),
+            PendingPayable::new(wallet_2.clone(), hash_2),
+            PendingPayable::new(wallet_3.clone(), hash_3),
+            PendingPayable::new(wallet_4.clone(), hash_4),
+        ];
+        let pending_payables_ref = pending_payables_owned
+            .iter()
+            .collect::<Vec<&PendingPayable>>();
+        let pending_payable_dao = PendingPayableDaoMock::new().fingerprints_rowids_result(vec![
+            (Some(4), hash_4),
+            (Some(1), hash_1),
+            (Some(3), hash_3),
+            (Some(2), hash_2),
+        ]);
+        let subject = PayableScannerBuilder::new()
+            .pending_payable_dao(pending_payable_dao)
+            .build();
+
+        let (existent, nonexistent) = subject
+            .separate_id_triples_by_existent_and_nonexistent_fingerprints(&pending_payables_ref);
+
+        assert_eq!(
+            existent,
+            vec![
+                PendingPayableTriple::new(&wallet_1, hash_1, Some(1)),
+                PendingPayableTriple::new(&wallet_2, hash_2, Some(2)),
+                PendingPayableTriple::new(&wallet_3, hash_3, Some(3)),
+                PendingPayableTriple::new(&wallet_4, hash_4, Some(4))
+            ]
+        );
+        assert!(nonexistent.is_empty())
+    }
+
+    #[test]
+    fn symmetry_check_happy_path() {
+        let hash_1 = make_tx_hash(123);
+        let hash_2 = make_tx_hash(456);
+        let hash_3 = make_tx_hash(789);
+        let blockchain_bridge_returned_pending_payables = vec![
+            PendingPayable::new(make_wallet("abc"), hash_1),
+            PendingPayable::new(make_wallet("def"), hash_2),
+            PendingPayable::new(make_wallet("ghi"), hash_3),
+        ];
+        let bb_returned_p_payables_ref = blockchain_bridge_returned_pending_payables
+            .iter()
+            .collect::<Vec<&PendingPayable>>();
+        let rowids_and_hashes_from_fingerprints =
+            vec![(Some(3), hash_1), (Some(5), hash_2), (Some(6), hash_3)];
+
+        PayableScanner::symmetry_check(
+            &bb_returned_p_payables_ref,
+            &rowids_and_hashes_from_fingerprints,
+        )
```

Comment:
If you'd like you can make this function return a boolean and maybe panic somewhere else. It would provide a strong test for the happy path. In case you want to reuse the code with panic, you may migrate the call to that function with a panic in an outboard function.